src/common/io.cc

```diff
@@ -384,6 +384,36 @@ int64_t read_buffer(int fd, int64_t *type, UT_array *buffer) {
   return 0;
 }
 
+int64_t read_vector(int fd, int64_t *type, std::vector<uint8_t> &buffer) {
+  int64_t version;
+  int closed = read_bytes(fd, (uint8_t *) &version, sizeof(version));
```

Comment:
I fixed it in the next PR, which modernizes some of the IO routines that the plasma store uses.